src/org/javarosa/core/model/condition/Constraint.java

```diff
@@ -81,6 +84,9 @@ private void attemptConstraintCompile() {
     public void readExternal(DataInputStream in, PrototypeFactory pf) throws IOException, DeserializationException {
         constraint = (IConditionExpr)ExtUtil.read(in, new ExtWrapTagged(), pf);
         constraintMsg = ExtUtil.nullIfEmpty(ExtUtil.readString(in));
+        if (constraintMsg != null) {
+            constraintMsg = constraintMsg.trim();
+        }
```

Comment:
There’s no way to write an untrimmed constraintMsg so this will never do anything. Right?

Reply:
It's needed if you have for example a `constraintMsg` like in the test form
`constraintMsg=" jr:itext('/data/q1:constraintMsg')"`
with a space before `jr:itext`.

Reply:
Sorry, what I meant was the trimming happens in the Constraint constructor, so at this point in the code there’s no way the constraintMsg would need trimming.

Reply:
Trimming is needed in both (constructor and here) otherwise it doesn't work.

Reply:
If you’re reading constraintMsg with readExternal, it’s been written with writeExternal, right? In order to be written with writeExternal, constraintMsg must have arrived via the constructor, which trims it. Do you follow my reasoning?

Reply:
Ahhh yes you are right. Probably I opened the form for the first time without trimming in the constructor and that's why I thought it's needed after reading too. Fixed, thanks!